packages/http/src/types.ts

```diff
@@ -446,25 +439,59 @@ export interface HttpOperationResponse {
 
 export interface HttpOperationResponseContent {
   headers?: Record<string, ModelProperty>;
-  body?: HttpOperationResponseBody;
+  body?: HttpOperationBody | HttpOperationMultipartBody;
 }
 
-export interface HttpOperationBody {
-  /**
-   * Content types.
-   */
-  contentTypes: string[];
+export interface HttpOperationBodyBase {
+  /** Content types. */
+  readonly contentTypes: string[];
+}
 
-  /**
-   * Type of the operation body.
-   */
-  type: Type;
+export interface HttpBody {
+  readonly type: Type;
 
   /** If the body was explicitly set with `@body`. */
   readonly isExplicit: boolean;
 
   /** If the body contains metadata annotations to ignore. For example `@header`. */
   readonly containsMetadataAnnotations: boolean;
+
```

Comment:
Here, it feels like we should also have a mechanism for getting at the visibility-filtered properties of the body (if the body is a model type)

Reply:
yeah I agree but this feels out of scope of this PR